scipy/signal/_short_time_fft.py

```diff
@@ -773,12 +775,29 @@ def from_win_equals_dual(
             SFT._scaling = scale_to
         return SFT
 
+    @staticmethod
+    def _make_elements_readonly(a):
+        """Helper to make the array `a` immutable.
+
+        This function is used only for making the attributes `win` and `dual_win`
+        immutable by setting the flag WRITEABLE of `a` to ``False``.
+        This prevents the following:
+
+        >>> SFT = ShortTimeFFT(np.ones(6), 3, fs=1)  # doctest: +SKIP
+        ... SFT.win[3] = 5  # this raises a ValueError
+
+        This one-line helper exists to make future porting to Array-API compatbile
+        behavior easier.
+        """
+        a.setflags(write=False)
```

Comment:
Up to you but IMO probably not worth a static method for a one liner

Reply:
FWIW, I'm +1 to what Jake said. 

If you *really* want to make array attributes immutable, the only future-proof way is to make them properties where getters return copies:

```
@property
def win(self):
    return self._win.copy()
```

Whether this is worth it vs documenting that users should not modify them willy-nilly... My knee-jerk reaction is towards the latter but maybe it is really justified in this particular case, I don't know.

Reply:
* Using `.copy()` is not too attractive, since `win` and `dual_win` are used internally quite a bit amd `dual_win` is created lazily. Furthermore, a design idea of the `ShortTimeFFT` class was to provide all STFT variables (like `win` or `f`) with little or no overhead.
* You're right, a one-line helper is probably not that useful → removed.
* I do think that un-setting the WRITABLE flag is marginally more useful than not doing it. Furthermore, this can be reversed in the future without breaking any existing code.
* I squashed the commits, because the back and forth is not useful without the context of this discussion.